src/codec/builtin/ListCNFixedSizeCodec.ts

```diff
@@ -18,37 +17,35 @@ export class ListCNFixedSizeCodec {
     ): T[] {
         const type = FixSizedTypesCodec.decodeByte(frame.content, 0);
         const count = FixSizedTypesCodec.decodeInt(frame.content, 1);
+        let position = ListCNFixedSizeCodec.HEADER_SIZE;
+        let readCount = 0;
 
-        const res = [];
+        const res = new Array(count);
         switch (type) {
             case ListCNFixedSizeCodec.TYPE_NULL_ONLY:
                 for (let i = 0; i < count; i++) {
-                    res.push(null);
+                    res[i] = null;
                 }
                 break;
             case ListCNFixedSizeCodec.TYPE_NOT_NULL_ONLY:
                 for (let i = 0; i < count; i++) {
-                    res.push(decoder(frame.content, ListCNFixedSizeCodec.HEADER_SIZE + i * itemSizeInBytes));
+                    res[i] = decoder(frame.content, ListCNFixedSizeCodec.HEADER_SIZE + i * itemSizeInBytes);
                 }
                 break;
             default:
-                assert.strictEqual(type, ListCNFixedSizeCodec.TYPE_MIXED, 'Invalid type');
-                let position = ListCNFixedSizeCodec.HEADER_SIZE;
-                let readCount = 0;
                 while (readCount < count) {
                     const bitmask = FixSizedTypesCodec.decodeByte(frame.content, position);
                     for (let i = 0; i < ListCNFixedSizeCodec.ITEMS_PER_BITMASK && readCount < count; i++) {
                         const mask = 1 << i;
                         if ((bitmask & mask) === mask) {
-                            res.push(decoder(frame.content, position));
+                            res[i] = decoder(frame.content, position);
```

Comment:
I don't think this works, this will cycle between the first 8 elements of the result and overwrite elements. We should be using the readCount as the index